### PR TITLE
Add ball joint

### PR DESCRIPTION
**Pull request by vsamy**
Just the possibility to use joint with type=ball.

Type name is same as for sdfformat: www.sdformat.org/spec?ver=1.6&elem=joint

### CODE REVIEW

File: src/urdf.cpp [hunk 1]
```diff
@@ -112,10 +112,15 @@ rbd::Joint::Type rbdynFromUrdfJoint(const std::string & type, bool hasSphericalS
   else if(type == "floating")
   {
     if (hasSphericalSuffix)
-      return rbd::Joint::Spherical;
+      {
+        std::cerr << "hasSphericalSuffix is deprecated and will be deleted in future release. Please use type=ball instead (style of sdf format)" << std::endl;
```
Comment (gergondet) [inline src/urdf.cpp]: indentation is broken here,there is also no plan to deprecate spherical suffix in the future

**Comment by vsamy** [comment]
C'est mieux la ?

**Comment by gergondet** [comment]
Yup! Merging when CI lands.

Thanks!

**Comment by vsamy** [comment]
Cool thanks :)

**Pull request merged**